src/runtime-rs/crates/resource/src/share_fs/sandbox_bind_mounts.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Note:
+// sandbox_bind_mounts supports kinds of mount patterns, for example:
+// (1) "/path/to", with default readonly mode.
+// (2) "/path/to:ro", same as (1).
+// (3) "/path/to:rw", with readwrite mode.
+//
+// sandbox_bind_mounts: ["/path/to", "/path/to:rw", "/mnt/to:ro"]
+//
+
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use anyhow::{anyhow, Context, Result};
+
+use super::utils::{do_get_host_path, mkdir_with_permissions};
+use kata_sys_util::{fs::get_base_name, mount};
+use kata_types::config::TomlConfig;
+
+const SANDBOX_BIND_MOUNTS_DIR: &str = "sandbox-mounts";
+const SANDBOX_BIND_MOUNTS_RO: &str = ":ro";
+const SANDBOX_BIND_MOUNTS_RW: &str = ":rw";
+
+#[derive(Clone, Default, Debug)]
+pub struct SandboxBindMounts {
+    sid: String,
+    toml_config: Arc<TomlConfig>,
+}
+
+impl SandboxBindMounts {
+    pub async fn new(sid: String, toml_config: Arc<TomlConfig>) -> Result<Self> {
+        Ok(SandboxBindMounts { sid, toml_config })
+    }
+
+    fn parse_sandbox_bind_mounts<'a>(&self, bind_mnt_src: &'a str) -> Result<(&'a str, &'a str)> {
+        // get the bindmount's r/w mode
+        let bind_mounts_rw = if bind_mnt_src.ends_with(SANDBOX_BIND_MOUNTS_RW) {
+            SANDBOX_BIND_MOUNTS_RW
+        } else {
+            SANDBOX_BIND_MOUNTS_RO
+        };
+
+        // get the true bindmount from the string
+        let sb_bind_mnt = bind_mnt_src.trim_end_matches(bind_mounts_rw);
+
+        Ok((bind_mounts_rw, sb_bind_mnt))
+    }
+
+    pub async fn setup_sandbox_bind_mounts(&self) -> Result<()> {
+        if self.toml_config.runtime.sandbox_bind_mounts.is_empty() {
+            info!(sl!(), "sandbox_bind_mounts is empty, skip it.");
+            return Ok(());
+        }
+
+        // /run/kata-containers/shared/sandboxes/<sid>/rw/passthrough/sandbox-mounts
+        let host_mounts_target =
+            do_get_host_path(SANDBOX_BIND_MOUNTS_DIR, self.sid.as_str(), "", true, false);
+        let host_mounts_path = Path::new(host_mounts_target.as_str());
+
+        let mut mounted_list: Vec<PathBuf> = Vec::new();
+        let mut mounted_map: HashMap<String, bool> = HashMap::new();
+        for src in &self.toml_config.runtime.sandbox_bind_mounts {
+            let (bind_mounts_rw, second) = self
```

Comment:
why the return value here is called `second`? From `parse_sandbox_bind_mounts `, it should called `sb_bind_mnt `?

`second` seems less strightforword for me.

Reply:
Thx.
One clear name bindmount replace of `second`